src/expr/src/scalar.proto

```diff
@@ -736,7 +736,9 @@ message ProtoDomainLimit {
 message ProtoEvalError {
     message ProtoUnsupported {
         string feature = 1;
-        optional uint64 issue_no = 2;
```

Comment:
We need to be careful with this change as it can prevent data from being compacted in persist. We should determine if we have any of these errors durably recorded in persist before merging the change.

Reply:
These are not used in the catalog persist shard. Would they be used anywhere else?

Reply:
We do write the error message text to `mz_internal.mz_statement_execution_history` (the activity log) in some cases. Would this change make it so the index on the activity log wouldn't compact properly?

Reply:
I think we can write the error proto itself to *any* persist shard, but I'm not exactly sure about this variant. Someone who knows more would need to check.

The issue is that once we fix the error, we'd generate a retraction, but that retraction would not match any durable error, so both the insert and retraction would stay around forever. The same problem happens for the in-memory representation I think and can cause problems at runtime.

Reply:
In this case, changing the name of the field won't change its on-disk encoding, so we can do that to get around the problem. @chaas can you remove the `reserved 2` and bumping the field number, and just do a rename of the existing field with ID 2?

Reply:
I think I successfully confused myself! For the in-memory representation, it shouldn't cause problems because the persist sink self-corrects the contents of a materialized view. Only the persist compaction might leave things around that are semantically equivalent but have a different bit pattern. So, retracting my request changes!